boxoffice/templates/boxoffice.js

```diff
@@ -210,6 +211,59 @@ $(function() {
           boxoffice.ractive.set('activeTab', boxoffice.ractive.get('tabs.selectItems.id'));
           boxoffice.ractive.scrollTop();
         },
+        applyDiscount: function(discountCoupons) {
+          //Ask server for the corresponding line_item for the discount coupon. Add one quantity of that line_item
+          $.post({
+            url: boxoffice.config.resources.kharcha.urlFor(),
+            crossDomain: true,
+            dataType: 'json',
+            headers: {'X-Requested-With': 'XMLHttpRequest'},
+            contentType: 'application/json',
+            data: JSON.stringify({
+              line_items: boxoffice.ractive.get('order.line_items').map(function(line_item) {
+                return {
+                  quantity: 1,
+                  item_id: line_item.item_id
+                };
+              }),
+              discount_coupons: discountCoupons
+            }),
+            timeout: 5000,
+            retries: 5,
+            retryInterval: 5000,
+            success: function(data) {
+              var applyDiscount;
+              var line_items = boxoffice.ractive.get('order.line_items');
+              line_items.forEach(function(line_item) {
+                if (data.line_items.hasOwnProperty(line_item.item_id)) {
+                  if(data.line_items[line_item.item_id].discounted_amount && line_item.quantity_available > 0) {
+                    applyDiscount = true;
+                    line_item.show_discountprice = true;
+                    line_item.discount_price = data.line_items[line_item.item_id].final_amount;
+                    line_item.discount_policies.forEach(function(discount_policy){
+                      if (data.line_items[line_item.item_id].discount_policy_ids.indexOf(discount_policy.id) >= 0) {
+                        discount_policy.applyCoupon = true;
+                      }
+                    });
+                  }
+                }
+              });
+              if(applyDiscount) {
```

Comment:
Do we really need this var? We can just check for `line_item.show_discountprice`?